### PR TITLE
Refactor actix handlers to use Extractor

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: graphql/src/lib.rs [hunk 1]
```diff
@@ -154,3 +103,42 @@ pub fn run_server(server_data: ServerData, addr: SocketAddr) -> Result<Server> {
...
 }
+
+struct SessionAndHandler {
+    pub session: u32,
+    pub handler: Arc<dyn HandleGraphQlRequest>,
+}
+
+impl FromRequest for SessionAndHandler {
+    type Error = Error;
+    type Future = Pin<Box<dyn Future<Output = Result<Self, Self::Error>>>>;
```
Comment (majecty) [inline graphql/src/lib.rs:114]: I don't know well about this type `Pin<Box<dyn Future`. I found an example and followed.

Comment (majecty) [reply graphql/src/lib.rs:114]: https://stackoverflow.com/questions/63308246/how-to-use-async-code-in-actix-web-extractors

Comment (junha1) [reply graphql/src/lib.rs:114]: AFAIK returning associated Future object is workaround of async trait. http://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/

File: graphql/src/lib.rs [hunk 2]
```diff
@@ -18,18 +18,18 @@ mod handler;
 
 use actix_web::{
```
Comment (junha1) [inline graphql/src/lib.rs:19]: Where does the term `Extractor` come from?

File: graphql/src/lib.rs [hunk 3]
```diff
@@ -154,3 +103,42 @@ pub fn run_server(server_data: ServerData, addr: SocketAddr) -> Result<Server> {
     });
     Ok(server.bind(addr)?.run())
 }
+
+struct SessionAndHandler {
+    pub session: u32,
```
Comment (junha1) [inline graphql/src/lib.rs:108]: We have `SessionId`

File: graphql/src/lib.rs (hunk 2 above)
Comment (majecty) [reply graphql/src/lib.rs:19]: From actix guide. They call `impl FromRequest` as extractor: https://actix.rs/docs/extractors/

Comment (junha1) [reply graphql/src/lib.rs:19]: Nice

File: graphql/src/lib.rs (hunk 1 above)
Comment (majecty) [reply graphql/src/lib.rs:114]: Thanks!

File: graphql/src/lib.rs (hunk 3 above)
Comment (majecty) [reply graphql/src/lib.rs]: Thanks, I changed the code to use it.

**Pull request merged**